lib/spack/spack/util/compression.py

```diff
@@ -288,17 +546,26 @@ def strip_extension(path):
     return path
 
 
+def check_extension(path, ext):
+    """Check if extension is present in path"""
+    # Strip sourceforge suffix.
```

Comment:
I think this comment is out of date

Reply:
Without that extra logic, this ends up being the same as `ext in path`

Reply:
The methods have been consolidated in a way that I believe addresses this comment.